feat: add support for accessor arrays and refactor `stats/base/max-by`

gururaj1512 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: na
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #None.

## Description

> What is the purpose of this pull request?

This pull request:

-   update `stats/base/max-by` to follow latest project conventions
-   add support for accessor arrays

## Related Issues

> Does this pull request have any related issues?

This pull request:

-  No

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gururaj1512 committed:
feat: add support for accessor arrays and refactor `stats/base/max-by`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/max-by/lib/accessors.js:
```diff
@@ -27,60 +27,68 @@ var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
+	get = x.accessors[ 0 ];
+
+	if ( N === 1 || strideX === 0 ) {
+		v = clbk.call( thisArg, get( xbuf, offsetX ), 0, 0, x );
```
```suggestion
		v = clbk.call( thisArg, get( xbuf, offsetX ), 0, offsetX, x );
```

kgryte committed:
fix: use correct strided index

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/max-by/lib/ndarray.js:
```diff
@@ -47,36 +49,41 @@ var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
+		return accessors( N, o, strideX, offsetX, clbk, thisArg );
+	}
+	if ( N === 1 || strideX === 0 ) {
+		v = clbk.call( thisArg, x[ offsetX ], 0, 0, x );
```
```suggestion
		v = clbk.call( thisArg, x[ offsetX ], 0, offsetX, x );
```

kgryte committed:
fix: provide correct stride index

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/max-by/test/test.ndarray.js:
```diff
@@ -173,14 +279,31 @@ tape( 'if provided a `stride` parameter equal to `0`, the function returns the f
+
 tape( 'the function supports an offset parameter', function test( t ) {
 	var x;
 	var v;
```
@gururaj1512 Missing two sets of tests. 1) supporting an offset parameter for accessor arrays. 2) supporting a callback execution context when providing an accessor array.

kgryte reviewed (changes requested):
Other than a couple of missing tests, this looks good.

gururaj1512 committed:
test: add accessor tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM
